playbooks/lets-encrypt.yml

```diff
@@ -0,0 +1,58 @@
+---
+# lets-encrypt.yml asks questions about Streisand domain name, which is used to
+# request TLS certificates from Let's Encrypt server. If user fails to answer
+# these questions, certain roles are disabled.
+- name: Collect information about the Streisand domain
+# =========================================
+  hosts: streisand-host
+  gather_facts: yes
+
+  vars_prompt:
+    - name: "streisand_domain_var"
+      prompt: |
+        Which domain do you want to use to host Streisand server?
+
+        This is an optional question. If you have a domain that points to your
+        Streisand server, the installation scripts can request Let's Encrypt
+        HTTPS certificates for you automatically.  If you do not provide one or
+        the request fails, self-signed certificates will be used instead.
+
+        If you have just created a new cloud server in previous steps, it is
+        good time to point your domain to your server's public address, as DNS
+        changes take time to propagate.
+
+        Please type your domain below. Press enter to skip.
+      private: no
+
+    - name: "streisand_admin_email_var"
+      prompt: |
+        Which email address do you want to use as contact of Streisand server?
+
+        This is an optioanl question. The email is only used as contact method
+        of your Let's Encrypt account.
+
+        Please type your email below. Press enter to use hostmaster email of Streisand domain.
+      default: "hostmaster@{{ streisand_domain_var }}"
+      private: no
+
+  pre_tasks:
+    - name: Set Streisand domain
+      set_fact:
+        streisand_domain: "{{ streisand_domain_var }}"
```

Comment:
I would love to see some basic checks here that the domain resolves to an IP (and maybe even that it's == the streisand ipv4 address we know the server is using). I think that will cut down on the most obvious failures right out of the box.

Reply:
@cpu May I ask if you are worrying about reaching Let's Encrypt's rate limit? According to acmetool's [user guide](https://hlandau.github.io/acme/userguide#challenge-completion-philosophy), self-tests are already done for HTTP-based validation before sending requests to Let's Encrypt for real.

Reply:
A little bit about rate limits and a little bit about user experience. It's always better to find out something is broken sooner than later.

> According to acmetool's user guide, self-tests are already done for HTTP-based validation before sending requests to Let's Encrypt for real.

Ok, let's call that sufficient for now. :-)